src/main/scala/io/qbeast/spark/internal/sources/v2/QbeastPartitionReader.scala

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.spark.internal.sources.v2
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.connector.read.PartitionReader
+
+/**
+ * Implementation of [[PartitionReader]].
+ */
+class QbeastPartitionReader private[v2] (private val iterator: Iterator[InternalRow])
```

Comment:
This partition reader is only Rows, not for Columnar Reading (which is the one that has better performance). We have to implement both a reader that PartitionReader[InternalRow] and one that does PartitionReader[ColumnBatch]. So, it would probably be best to call this class QbeastPartitionRowReader, or something similar

Reply:
There is trait called `PartitionReaderFactory` that has both methods: createColumnarReader and `createReader` (and `supportColumnarReads` to verify it can read in columnar batches). We can extend that.